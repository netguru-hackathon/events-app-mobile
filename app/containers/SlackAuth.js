import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Platform,
  WebView,
} from 'react-native';
import url from 'url';
import UserAgentIOS from 'rn-ios-user-agent';

class SlackAuth extends Component {
  static propTypes = {
    isStarted: PropTypes.bool.isRequired,
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
      state: PropTypes.shape({
        params: PropTypes.shape({
          authURL: PropTypes.string.isRequired,
        }),
      }),
    }).isRequired,
  }

  componentWillMount() {
    if (Platform.OS === 'ios') { UserAgentIOS.set('Mozilla/5.0 Google'); }
  }

  webViewRequest = (event) => {
    const parsedUrl = url.parse(event.url, true);
    const { navigation, isStarted } = this.props;

    if (isStarted) return false;
    if (parsedUrl.hostname === 'localhost' && parsedUrl.query.error) {
      navigation.navigate('Login');
      return false;
    }
    if (parsedUrl.hostname === 'localhost' && parsedUrl.query.code) {
      navigation.navigate('AccessTokenFetcher', { code: parsedUrl.query.code });
      this.vebview.stopLoading();
      return false;
    }
    return true;
  }

  render() {
    const { params } = this.props.navigation.state;

    return (
      <WebView
        ref={(c) => { this.vebview = c; }}
        source={{ uri: params.authURL }}
        userAgent={'Mozilla/5.0 Google'}
        onShouldStartLoadWithRequest={this.webViewRequest}
        onNavigationStateChange={this.webViewRequest}
      />
    );
  }
}

const mapStateToProps = state => ({
  isStarted: state.auth.isStarted,
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(SlackAuth);
