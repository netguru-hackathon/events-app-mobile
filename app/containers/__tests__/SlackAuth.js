import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SlackAuth from '../SlackAuth';

// https://github.com/facebook/react-native/issues/12440
jest.unmock('ScrollView');

test('renders correctly', () => {
  const tree = renderer.create(
    <SlackAuth
      navigation={{
        navigate: () => {},
        state: {
          params: {
            authURL: 'http://google.com',
          },
        },
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
