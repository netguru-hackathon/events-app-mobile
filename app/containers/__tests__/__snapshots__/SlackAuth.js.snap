// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript={undefined}
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
  />
</View>
`;
